diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 7dd5b5b..5baf6e2 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -142,3 +142,6 @@

/* http://elinux.org/CircuitCo:Basic_Proto_Cape */
/* #include "am335x-bone-basic-proto-cape.dtsi" */
+
+/* AKM sensor */
+#include "am335x-bone-akm-sensors.dtsi"
diff --git a/drivers/iio/magnetometer/Kconfig b/drivers/iio/magnetometer/Kconfig
index d86d226..1a6e527 100644
--- a/drivers/iio/magnetometer/Kconfig
+++ b/drivers/iio/magnetometer/Kconfig
@@ -16,6 +16,31 @@ config AK8975
To compile this driver as a module, choose M here: the module
will be called ak8975.

+config IIO_AKM_MAGN_AK0991X
+   tristate "Asahi Kasei AK0991X 3-Axis Magnetometer"
+   select IIO_BUFFER
+   select IIO_TRIGGERED_BUFFER
+   help
+     Say yes here to build support for Asahi Kasei AK09911/AK09912
+     3-Axis Magnetometer.
+
+     To compile this driver as a module, choose M here: the module
+     will be called ak0991x.
+
+config IIO_AKM_MAGN_AK0991X_I2C
+   tristate "support I2C bus connection"
+   depends on IIO_AKM_MAGN_AK0991X && I2C
+   default y
+   help
+     Say Y here if you have AK0991X hooked to an I2C bus.
+
+config IIO_AKM_MAGN_AK0991X_SPI
+   tristate "support SPI bus connection"
+   depends on IIO_AKM_MAGN_AK0991X && SPI
+   default y
+   help
+     Say Y here if you have AK0991X hooked to an SPI bus.
+
config MAG3110
tristate "Freescale MAG3110 3-Axis Magnetometer"
depends on I2C
diff --git a/drivers/iio/magnetometer/Makefile b/drivers/iio/magnetometer/Makefile
index 0f5d3c9..3a60aab 100644
--- a/drivers/iio/magnetometer/Makefile
+++ b/drivers/iio/magnetometer/Makefile
@@ -4,6 +4,9 @@

# When adding new entries keep the list in alphabetical order
obj-$(CONFIG_AK8975)    += ak8975.o
+obj-$(CONFIG_IIO_AKM_MAGN_AK0991X) += ak0991x-iio.o
+obj-$(CONFIG_IIO_AKM_MAGN_AK0991X_I2C) += ak0991x-i2c.o
+obj-$(CONFIG_IIO_AKM_MAGN_AK0991X_SPI) += ak0991x-spi.o
obj-$(CONFIG_MAG3110)   += mag3110.o
obj-$(CONFIG_HID_SENSOR_MAGNETOMETER_3D) += hid-sensor-magn-3d.o
